pkg/kotsadm/hostnetwork/hostnet_util.go

```diff
@@ -0,0 +1,77 @@
+package hostnetwork
+
+import v1 "k8s.io/api/core/v1"
+
+type PortMapping struct {
+	MinioMinio        int32
+	PostgresPostgres  int32
+	KotsadmKotsadmAPI int32
+	KotsadmKotsadm    int32
+}
+
+var (
+	hostnetPorts = PortMapping{
+		MinioMinio:        9000,
+		PostgresPostgres:  5432,
+		KotsadmKotsadm:    3000,
+		KotsadmKotsadmAPI: 3000, // todo this conflicts, need to fix which port the container exposes. Maybe there's an env var.
+	}
+	containerPorts = PortMapping{
+		MinioMinio:        9000,
+		PostgresPostgres:  5432,
+		KotsadmKotsadm:    3000,
+		KotsadmKotsadmAPI: 3000,
+	}
+)
+
+// Return a port map with either all zeroes (do not set HostPort fields)
+// or one with specific ports for each service we ship
+func HostPorts(useHostNetwork bool) PortMapping {
+	if useHostNetwork {
+		return hostnetPorts
+	}
+	return PortMapping{}
+}
+
+// Return a port map with either all zeroes (do not set HostPort fields)
+// or one with specific ports for each service we ship
+func ContainerPorts(useHostNetwork bool) PortMapping {
+	if useHostNetwork {
+		return hostnetPorts
+	}
+	return containerPorts
```

Comment:
doesn't this function always return the same data, since `hostnetPorts == containerPorts`? Is it meant to return all 0s if useHostNetwork is true?

Reply:
I will at some point change this line to 3001 at which point having two will make sense:

https://github.com/replicatedhq/kots/pull/299/files/1a5a54420c0d05f9ae757a5780f6f386dd04e465#diff-3c90b99d819355eb3992cf718b045328R17

Reply:
It might be possible to map a 3000 in-container to 3001 on-the-host, but I can't see how to do that. The docs says "HostPort must match ContainerPort if HostNetwork is set to true" which makes sense since we don't have any bridge to do translation.